src/main/java/org/broadinstitute/dsm/util/ElasticSearchUtil.java

```diff
@@ -1062,6 +1100,23 @@ else if (nameValue[0].startsWith(INVITATIONS)) {
                                 queryBuilder.must(queryActivityAnswer);
                                 NestedQueryBuilder query = QueryBuilders.nestedQuery(ACTIVITIES, queryBuilder, ScoreMode.Avg);
                                 finalQuery.must(query);
+
+                                // search for queries for the same field:
+                                // if found and it is Range and it was inserted into Nested block then remove
+                                // this Range from former place and add it into must() block together with this
+                                // `activities.questionsAnswers.stableId`==[field_name] query.
+                                QueryBuilder tmpBuilder = findQueryBuilderForFieldName(finalQuery, surveyParam[1], queryPartsMap);
+                                if (tmpBuilder instanceof RangeQueryBuilder && parentNestedOfRangeBuilder != null) {
+                                    activityAnswer.must(tmpBuilder);
+                                    BoolQueryBuilder finalQueryCopy = new BoolQueryBuilder();
+                                    for (Iterator<QueryBuilder> iterator = finalQuery.must().iterator(); iterator.hasNext(); ) {
```

Comment:
I think using for each loop instead of old version of usage of for with `Iterator` is more flexible and cleaner.

Reply:
I agree: I tried to use forEach loop but encountered a problem that it gives error regarding the variable `parentNestedOfRangeBuilder` complaining that it is not final or effectively final. In case of iterator it doesn't complain.

Reply:
Strange, you are not changing value of `parentNestedOfRangeBuilder`, even more we don't have lambda expression or anonymous class to have capturing problem, did you try like this construction: 
```
for (QueryBuilder builder : finalQuery.must()) {
    if (!builder.equals(parentNestedOfRangeBuilder)) {
        finalQueryCopy.must(builder);
    }
}
```